src/library_fs.js

```diff
@@ -1053,9 +1053,11 @@ FS.staticInit();
         mode = 0;
       }
       var node;
+      var isDirPath;
       if (typeof path == 'object') {
```

Comment:
reading this makes me think we should avoid this type of polymorphic input types.

Reply:
Yeah it does complicate code. In a few places, it helps avoid wasteful roundtripping if we need to call an API that expects a path but we have a node. But we could have differently named methods e.g., `openPath` and `openNode` rather than one that tries to handle both. And there are a lot of spots where we already are being wasteful. For example, `lookupPath` probably ought to have a relative form where it takes a base directory node to start at.

Reply:
I think separating `openPath` from `openNode` would be a great idea.

Reply:
Do we make any backwards compatibility guarantees about the FS api?

Reply:
Thats a good question.  I there it is documented so at the very least we would want to document any changes in the changelog and update the docs.

Reply:
Can we merge this change as is and address this in a followup?